### PR TITLE
refactor: improve log when no repository found based on autodiscovery options

### PR DESCRIPTION
**Pull request by azro352**
## Changes

refactor: improve log when no repository found based on autodiscovery options

 Closes https://github.com/renovatebot/renovate/discussions/25289

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

Sample outputs 
```
config { autodiscoverTopics: [], autodiscoverNamespaces: undefined }
... This can be caused by the token's permissions 

=====
config { autodiscoverTopics: undefined, autodiscoverNamespaces: undefined }
... This can be caused by the token's permissions 

=====
config { autodiscoverTopics: [], autodiscoverNamespaces: [] }
... This can be caused by the token's permissions 

=====
config { autodiscoverTopics: [ 'foo' ], autodiscoverNamespaces: [] }
... This can be caused by the token's permissions, or the option autodiscoverTopics (set to 'foo') 

=====
config { autodiscoverTopics: [], autodiscoverNamespaces: [ 'foo', 'foo' ] }
... This can be caused by the token's permissions, or the option autodiscoverNamespaces (set to 'foo,foo') 

=====
config {
  autodiscoverTopics: [ 'foo' ],
  autodiscoverNamespaces: [ 'foo', 'foo' ]
}
... This can be caused by the token's permissions, or the options autodiscoverTopics (set to 'foo') or autodiscoverNamespaces (set to 'foo,foo') 
```

### CODE REVIEW

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
I'll let the maintainers review this.

The PR should also have another prefix, it's not a `docs` type change.

**Review by rarkins — CHANGES_REQUESTED** [review]
I would prefer to keep it simpler, like adding "or the configured autodiscover topics/namespaces settings" and show that if either setting is truthy

**Comment by azro352** [comment]
@rarkins Do you want

1. exact values

```js
logger.debug({autodiscoverTopics: config.autodiscoverTopics, autodiscoverNamespaces: config.autodiscoverNamespaces}, "The account ... token\'s permissions or the configured autodiscover topics/namespaces settings");
```

2. defined replacement


```js
logger.debug(`The account ... token\'s permissions or the configured autodiscover topics (${config.autodiscoverTopics?.length ? "defined": "empty"})/namespaces (${config.autodiscoverNamespaces?.length ? "defined": "empty"}) settings`);
```

**Comment by rarkins** [comment]
@azro352 I needed to add better logging to autodiscover for other reasons today so created #25304. I think it might be sufficient to cover your needs too?

**Pull request closed without merging**
